### PR TITLE
Add tests for TiebreakerCore

### PR DESCRIPTION
**Pull request by rkolpakov**
Add tests for checkSealableIsPaused and checkProposalExists on TiebreakerCore

### CODE REVIEW

File: test/unit/committees/TiebreakerCore.t.sol [hunk 1]
```diff
@@ -201,6 +201,54 @@ contract TiebreakerCoreUnitTest is UnitTest {
...
+        assertEq(executionQuorum, quorum);
+        assertEq(quorumAt, Timestamp.wrap(0));
+        assertFalse(isExecuted);
+
+        vm.prank(committeeMembers[1]);
+        tiebreakerCore.sealableResume(sealable, nonce);
+
+        (support, executionQuorum, quorumAt, isExecuted) = tiebreakerCore.getSealableResumeState(sealable, nonce);
+        Timestamp quorumAtExpected = Timestamp.wrap(uint40(block.timestamp));
+        assertEq(support, quorum);
```
Comment (Psirex) [inline test/unit/committees/TiebreakerCore.t.sol:231]:
```suggestion
        assertEq(support, 2);
```

File: test/unit/committees/TiebreakerCore.t.sol [hunk 2]
```diff
@@ -201,6 +201,54 @@ contract TiebreakerCoreUnitTest is UnitTest {
...
+        assertEq(support, 1);
+        assertEq(executionQuorum, quorum);
+        assertEq(quorumAt, Timestamp.wrap(0));
+        assertFalse(isExecuted);
+
+        vm.prank(committeeMembers[1]);
+        tiebreakerCore.sealableResume(sealable, nonce);
+
+        (support, executionQuorum, quorumAt, isExecuted) = tiebreakerCore.getSealableResumeState(sealable, nonce);
+        Timestamp quorumAtExpected = Timestamp.wrap(uint40(block.timestamp));
```
Comment (Psirex) [inline test/unit/committees/TiebreakerCore.t.sol]:
```suggestion
        Timestamp quorumAtExpected = Timestamps.now();
```

File: test/unit/committees/TiebreakerCore.t.sol [hunk 3]
```diff
@@ -201,6 +201,54 @@ contract TiebreakerCoreUnitTest is UnitTest {
...
 
+    function test_getSealableResumeState_HappyPath() external {
+        uint256 nonce = tiebreakerCore.getSealableResumeNonce(sealable);
+        _mockSealableResumeSinceTimestamp(sealable, block.timestamp + 1000);
+
+        (uint256 support, uint256 executionQuorum, Timestamp quorumAt, bool isExecuted) =
+            tiebreakerCore.getSealableResumeState(sealable, nonce);
+        assertEq(support, 0);
+        assertEq(executionQuorum, quorum);
+        assertEq(quorumAt, Timestamp.wrap(0));
```
Comment (Psirex) [inline test/unit/committees/TiebreakerCore.t.sol]:
```suggestion
        assertEq(quorumAt, Timestamps.ZERO);
```

File: test/unit/committees/TiebreakerCore.t.sol (hunk 1 above)
Comment (Psirex) [reply test/unit/committees/TiebreakerCore.t.sol:231]: Or assertion that `quorum` is equal to 2 should be added

File: test/unit/committees/TiebreakerCore.t.sol [hunk 4]
```diff
@@ -201,6 +201,54 @@ contract TiebreakerCoreUnitTest is UnitTest {
...
+        assertFalse(isExecuted);
+
+        vm.prank(committeeMembers[2]);
+        tiebreakerCore.executeSealableResume(sealable);
+
+        (support, executionQuorum, quorumAt, isExecuted) = tiebreakerCore.getSealableResumeState(sealable, nonce);
+        assertEq(support, quorum);
+        assertEq(executionQuorum, quorum);
+        assertEq(quorumAt, quorumAtExpected);
+        assertTrue(isExecuted);
```
Comment (Psirex) [inline test/unit/committees/TiebreakerCore.t.sol:251]: Also required to check that after setting the new quorum value for the committee contract, the `getSealableResumeState()` still returns correct data

File: test/unit/committees/TiebreakerCore.t.sol [hunk 5]
```diff
@@ -240,11 +288,79 @@ contract TiebreakerCoreUnitTest is UnitTest {
         assertTrue(isExecuted);
     }
 
+    function testFuzz_checkProposalExists_HappyPath(uint256 existentProposalId) external {
```
Comment (Psirex) [inline test/unit/committees/TiebreakerCore.t.sol]: This fuzz test always hits the edge case where `proposalsCount()` equals `existentProposalId`. Instead of using the same fuzzed value for both proposalsCount and proposalId, use separate variables, and apply `vm.assume()` to filter out only valid (positive) combinations.

File: test/unit/committees/TiebreakerCore.t.sol [hunk 6]
```diff
@@ -240,11 +288,79 @@ contract TiebreakerCoreUnitTest is UnitTest {
...
+        vm.assume(existentProposalId > 0);
+
+        EmergencyProtectedTimelockMock(payable(emergencyProtectedTimelock)).setProposalsCount(existentProposalId);
+        this.external__checkProposalExists(existentProposalId);
+    }
+
+    function testFuzz_checkProposalExists_RevertOn_ProposalDoesNotExist(uint256 notExistentProposalId) external {
```
Comment (Psirex) [inline test/unit/committees/TiebreakerCore.t.sol]: Here is similar situation, but the proposals count always set to 0

File: test/unit/committees/TiebreakerCore.t.sol [hunk 7]
```diff
@@ -240,11 +288,79 @@ contract TiebreakerCoreUnitTest is UnitTest {
...
+        vm.expectRevert(abi.encodeWithSelector(TiebreakerCoreCommittee.InvalidSealable.selector, address(0)));
+        this.external__checkSealableIsPaused(address(0));
+    }
+
+    function testFuzz_checkSealableIsPaused_RevertOn_SealableNotPaused(
+        Timestamp currentTimestamp,
+        Timestamp resumeSinceTimestamp
+    ) external {
+        vm.assume(resumeSinceTimestamp.toSeconds() >= 0);
+        vm.assume(currentTimestamp.toSeconds() > resumeSinceTimestamp.toSeconds());
```
Comment (Psirex) [inline test/unit/committees/TiebreakerCore.t.sol]: No need to call `.toSeconds()` here

File: test/unit/committees/TiebreakerCore.t.sol [hunk 8]
```diff
@@ -240,11 +288,79 @@ contract TiebreakerCoreUnitTest is UnitTest {
...
+    function test_checkSealableIsPaused_RevertOn_ZeroAddress() external {
+        vm.expectRevert(abi.encodeWithSelector(TiebreakerCoreCommittee.InvalidSealable.selector, address(0)));
+        this.external__checkSealableIsPaused(address(0));
+    }
+
+    function testFuzz_checkSealableIsPaused_RevertOn_SealableNotPaused(
+        Timestamp currentTimestamp,
+        Timestamp resumeSinceTimestamp
+    ) external {
+        vm.assume(resumeSinceTimestamp.toSeconds() >= 0);
```
Comment (Psirex) [inline test/unit/committees/TiebreakerCore.t.sol]:
```suggestion
        vm.assume(resumeSinceTimestamp.isNotZero());
```

File: test/unit/committees/TiebreakerCore.t.sol [hunk 9]
```diff
@@ -240,11 +288,79 @@ contract TiebreakerCoreUnitTest is UnitTest {
...
+        vm.assume(currentTimestamp.toSeconds() > resumeSinceTimestamp.toSeconds());
+
+        vm.warp(currentTimestamp.toSeconds());
+        _mockSealableResumeSinceTimestamp(sealable, resumeSinceTimestamp.toSeconds());
+        vm.expectRevert(abi.encodeWithSelector(TiebreakerCoreCommittee.SealableIsNotPaused.selector, sealable));
+        this.external__checkSealableIsPaused(sealable);
+    }
+
+    function test__checkSealableIsPaused_RevertOn_SealableNotPaused_CurrentBlock() external {
```
Comment (Psirex) [inline test/unit/committees/TiebreakerCore.t.sol]: two underscoers in the name

File: test/unit/committees/TiebreakerCore.t.sol [hunk 10]
```diff
@@ -240,11 +288,79 @@ contract TiebreakerCoreUnitTest is UnitTest {
...
+
     function _mockSealableResumeSinceTimestamp(address sealableAddress, uint256 resumeSinceTimestamp) internal {
         vm.mockCall(
             sealableAddress,
             abi.encodeWithSelector(ISealable.getResumeSinceTimestamp.selector),
             abi.encode(resumeSinceTimestamp)
         );
     }
+
+    function external__checkSealableIsPaused(address sealableContract) public view {
```
Comment (Psirex) [inline test/unit/committees/TiebreakerCore.t.sol]: No need to make explicit external calls via `this.`. The `tiebreakerCore.checkSealableIsPaused()` will be sent as an external call

**Review by Psirex — CHANGES_REQUESTED** [review]

**Review by Psirex — APPROVED** [review]
LGTM👍🏻
